lib/classes/PluginManager.test.js

```diff
@@ -123,14 +124,14 @@ describe('PluginManager', () => {
     }
 
     functions() {
-      return new Promise((resolve) => {
+      return new BbPromise((resolve) => {
```

Comment:
I suggest to use BbPromisy.try() instead of the `new BbPromise` anti-pattern:

```js
return BbPromise.try(() => {
  this.deployedFunctions = this.deployedFunctions + 1;
})
```

This will also catch any synchronous throws and rejects your promise accordingly.